pyogrio/tests/test_geopandas_io.py

```diff
@@ -1927,6 +1952,8 @@ def test_read_dataset_kwargs(nested_geojson_file, use_arrow):
         geometry=[shapely.Point(0, 0)],
         crs="EPSG:4326",
     )
+    if GDAL_GE_311 and use_arrow:
```

Comment:
Can you please add a comment indicating why it is necessary to cast to `object` here?

Reply:
Yeah, this is the part that should go away based on the other TODO comment above. But will add a comment until then